README.md

```diff
@@ -99,16 +79,94 @@ end
 ## Specifying a default Facter version
 
 By default, `on_supported_os` will return the facts for the version of Facter
-that it has loaded (usually this is Facter 2.5.1). This behaviour can be
-overridden by setting the `default_facter_version` RSpec setting in your
-`spec/spec_helper.rb` file.
+that it has loaded.
+It will match the Facter version based on the loaded Puppet version to what the official packages provide.
+The mapping is
+stored in `ext/puppet_agent_components.json` (check the
+[maintenance](#maintenance) section for details) and computated in the
+`facter_version_for_puppet_version` method. This behaviour can be overridden by
+setting the `default_facter_version` RSpec setting in your `spec/spec_helper.rb`
+file.
```

Comment:
My direct feedback was lazy. I'd consistently use sentence case

Reply:
you mean one line per sentence? can update it.

Reply:
Updated the style, let me know what you think about it.